Sources/SwiftGRPCNIO/GRPCChannelHandler.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import SwiftProtobuf
+import NIO
+import NIOHTTP1
+
+// Processes individual gRPC messages and stream-close events on a HTTP2 channel.
+public protocol GRPCCallHandler: ChannelHandler {
+  func makeGRPCServerCodec() -> ChannelHandler
+}
+
+// Provides `GRPCCallHandler` objects for the methods on a particular service name.
+// Implemented by the generated code.
+public protocol CallHandlerProvider {
+  var serviceName: String { get }
+
+  // Looks up and returns the `GRPCCallHandler` for a particular method. Returns nil for unsupported methods.
+  func handleMethod(_ methodName: String, request: HTTPRequestHead, serverHandler: GRPCChannelHandler, channel: Channel) -> GRPCCallHandler?
+}
+
+// Listens on a newly-opened HTTP2 subchannel and waits for the request headers to become available.
+// Once those are available, asks the `CallHandlerProvider` corresponding to the request's service name for an
+// `GRPCCallHandler` object. That object is then forwarded the individual gRPC messages.
+public final class GRPCChannelHandler: ChannelInboundHandler {
+  public typealias InboundIn = RawGRPCServerRequestPart
+
+  public typealias OutboundOut = RawGRPCServerResponsePart
+
+  fileprivate let servicesByName: [String: CallHandlerProvider]
+
+  public init(servicesByName: [String: CallHandlerProvider]) {
+    self.servicesByName = servicesByName
+  }
+
+  public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
```

Comment:
Might be nice to put the protocol conformance functions (throughout this PR) in separate extensions in the same file to make it clearer as to where they're coming from

Reply:
Agreed; changed that on all handlers..